mesonbuild/interpreterbase/interpreterbase.py

```diff
@@ -169,22 +169,17 @@ def evaluate_codeblock(self, node: mparser.CodeBlockNode, start: int = 0, end: T
             return
         if not isinstance(node, mparser.CodeBlockNode):
             raise InvalidCode.from_node('Tried to execute a non-codeblock. Possibly a bug in the parser.', node=node)
-        statements = node.lines[start:end]
-        i = 0
-        while i < len(statements):
-            # NOTE: self.current_node is updated in self.evaluate_statement, while cur is not
-            cur = statements[i]
-            self.current_lineno = cur.lineno
+        for self.current_node in node.lines[start:end]:
+            self.current_lineno = self.current_node.lineno
             try:
-                self.evaluate_statement(cur)
+                self.evaluate_statement(self.current_node)
             except MesonException as e:
                 # Assume that if file is set, then the information in the exception is correct
                 if e.file is None:
                     e.update_position(self.current_node)
                 raise
             except Exception as e:
                 raise mesonlib.MesonExceptionWrapper.from_node(e, self.current_node)
-            i += 1 # In THE FUTURE jump over blocks and stuff.
```

Comment:
I have no idea what this comment is proposing to do but it seems unrelated to getting rid of `i`.

Reply:
The code dates back to commit 676dc51ac9324a7cb992eddc47ae94d941501c4a. Even more clueless what it means.

Reply:
That's really early in the development of the interpreter, it did not support if-block back then, only function call and assignment. We do jump over if-block nowadays.

Reply:
Oh great, so we get to drop a confusingly meaningless comment. Yay!

Reply:
My guess: @jpakkane's intention was to have i+=number_of_lines_in_the_block when a `if` evaluates to False. But the way it's implemented now is block statements are in the IfNode object, so that loop only have top level statements, there is nothing to jump over.